File: proj1/queue.cpp
#include <iostream>
#include "queue.h"


void QueueNode::push(int val)
{ 
    std::cout << "======================================" << std::endl;
    queue[rear] = val;
    std::cout << "Pushed " << val << " into position " << rear << "." << std::endl;
    rear++;
}

int QueueNode::pop()
{ 
    std::cout << "======================================" << std::endl;
    int result;

    result = queue[front];
    std::cout << "Popped " << result << " from position " << front << "." << std::endl;
    front++;
    return result;
}

void QueueNode::print() const
{
  std::cout << "===============PRINTING==QUEUE=============" << std::endl;
  int i = front;

  if(isEmpty()) {
    std::cout << "Queue is empty." << std::endl;
    return;
  }

  for(i; i<rear;i++)
  {
    std::cout << "Element at postion " << i << " is " << queue[i] << std::endl;
  }
}

File: proj2_original/Queue.java
public class Queue
{
        private int front;
        private int rear;
        
        private Vertex queue[];
        
        public Queue(int size)
        {
	queue = new Vertex[size+1];
	front = 0;
	rear = 0;
        }
        
        public void print()
        {
	int i;
	
	System.out.println("<<<<<<<<<<<<<<<<Printing queue...>>>>>>>>>>>>>>>>>>");
	for(i=0; i<queue.length ; i++)
	{
	        queue[i].print();
	}
	System.out.println("<<<<<<<<<<<<<<<<=================>>>>>>>>>>>>>>>>>>");
        }
        
        public void push(Vertex v)
        {
	if(front==queue.length-1)
	        System.out.println("==========QUEUE SIZE EXCEEDED!!===========");
	        
	queue[front]=v;
	
	System.out.println("Inserting <<" + v.getValue() + ">>.");
	
	front++;
        }
        
        public Vertex get()
        {
	Vertex result = queue[rear];
	
	System.out.println("Getting <<" + result.getValue() + ">>.");
	rear = rear + 1;
	return result;
        }
        
        public boolean isEmpty()
        {
	return (front==rear);
        }

        
}